src/devices/Mpr121/Mpr121.cs

```diff
@@ -0,0 +1,220 @@
+
+        private Dictionary<Channels, bool> _statuses;
+
+        private int _periodRefresh;
+
+        /// <summary>
+        /// Notifies about a the channel statuses have been changed.
+        /// Refresh period can be changed by setting PeriodRefresh property.
+        /// </summary>
+        public event EventHandler<ChannelStatusesChangedEventArgs> ChannelStatusesChanged;
+
+        /// <summary>
+        /// Gets ot sets the period in milliseconds to refresh the channels statuses.
+        /// </summary>
+        /// <remark>
+        /// Set value 0 to stop the automatically refreshing. Setting the value greater than 0 will start/update auto-refresh.
+        /// </remark>
+        public int PeriodRefresh
+        {
+            get { return _periodRefresh; }
+
+            set
+            {
+                _periodRefresh = value;
+
+                if (_periodRefresh > 0 && _timer == null)
+                {
+                    _timer = new Timer(RefreshChannelStatuses, this, TimeSpan.FromMilliseconds(_periodRefresh), TimeSpan.FromMilliseconds(_periodRefresh));
+                }
+                else if (_periodRefresh == 0 && _timer != null)
+                {
+                    _timer.Dispose();
+                    _timer = null;
+                }
+                else if (_periodRefresh > 0 && _timer != null)
+                {
+                    _timer.Change(TimeSpan.FromMilliseconds(_periodRefresh), TimeSpan.FromMilliseconds(_periodRefresh));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initialize a MPR121 controller.
+        /// </summary>
+        /// <param name="device">The i2c device.</param>
+        /// <param name="periodRefresh">The period in milliseconds of refresing the channel statuses.</param>
+        /// <param name="configuration">The controller configuration.</param>
+        public Mpr121(I2cDevice device, int? periodRefresh = null, Mpr121Configuration configuration = null)
+        {
+            _device = device;
+            periodRefresh = periodRefresh ?? 0;
+            configuration = configuration ?? GetDefaultConfiguration();
+
+            _statuses = new Dictionary<Channels, bool>();
+            foreach (Channels channel in Enum.GetValues(typeof(Channels)))
+            {
+                _statuses.Add(channel, false);
+            }
+
+            InitializeController(configuration);
+
+            PeriodRefresh = periodRefresh.Value;
+        }
+
+        public void Dispose()
+        {
+            if (_device != null)
+            {
+                _device.Dispose();
+                _device = null;
+            }
+
+            if (_timer != null)
+            {
+                _timer.Dispose();
+                _timer = null;
+            }
+        }
+
+        /// <summary>
+        /// Reads the channel statuses of MPR121 controller.
+        /// </summary>
+        public IReadOnlyDictionary<Channels, bool> ReadChannelStatuses()
```

Comment:
I don't know much about this device but does it make sense to just create:
`bool ReadChannelStatus(Channel channel)` or `GetChannelStatus`?

Reply:
Yes, this method could be useful. I added it with some remarks in xmldoc.